core/src/main/java/nu/yona/server/analysis/entities/ActivityCommentMessage.java

```diff
@@ -1,37 +1,46 @@
 /*******************************************************************************
- * Copyright (c) 2016, 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
- * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ * Copyright (c) 2016, 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
  *******************************************************************************/
 package nu.yona.server.analysis.entities;
 
+import java.util.Arrays;
+import java.util.List;
 import java.util.Set;
 import java.util.UUID;
 
 import javax.persistence.Entity;
+import javax.persistence.FetchType;
 import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
 import javax.persistence.OneToOne;
 
+import org.hibernate.annotations.BatchSize;
+
 import nu.yona.server.messaging.entities.BuddyMessage;
 import nu.yona.server.messaging.entities.Message;
 
 @Entity
+@BatchSize(size = 100)
 public class ActivityCommentMessage extends BuddyMessage
 {
 	@ManyToOne
 	private IntervalActivity intervalActivity;
 
 	/**
 	 * Buddy comment messages are always sent in pairs: a message to the buddy and a copy for the user. This is one of the two
-	 * properties that maintain the relationship between the two messages.
+	 * properties that maintain the relationship between the two messages.<br/>
+	 * NOTE: this field is defined as a OneToMany/List, as a technical trick to implement lazy loading. Practically, it's a
+	 * OneToOne relationship. This trick delivers a considerable performance benefit.
 	 */
-	@OneToOne(mappedBy = "buddyMessage")
-	private ActivityCommentMessage senderCopyMessage;
+	@OneToMany(mappedBy = "buddyMessage", fetch = FetchType.LAZY)
+	private List<ActivityCommentMessage> senderCopyMessageList;
```

Comment:
Suggestion: could be called `senderCopyMessageSingleton`.

Reply:
That gives me the wrong impression. It's the list and that's a trick, but I'd rather see that from the name.

Reply:
OK, I meant singleton as [in the mathematical sense](https://en.wikipedia.org/wiki/Singleton_%28mathematics%29) (a list with only one item). If you think that isn't very clear, can't we reflect in a different way in the name that we expect only one item?

Reply:
Now I understand what you meant. Most people will think of the [Singleton pattern](https://en.wikipedia.org/wiki/Singleton_pattern) and that's not what we have here. I'll rename it into ``senderCopyMessageHolder`` and create handling methods accordingly.